File: Beecrowd/2-AD-HOC/1091-Divisão_da_Nlogônia-1/1091.py
while True:
    repeat = int(input())
    if repeat == 0:
        break

    xlog, ylog = map(int, input().split())

    for _ in range(repeat):
        xcurrent, ycurrent = map(int, input().split())

        if xcurrent == xlog or ycurrent == ylog:
            print("divisa")
        elif xcurrent > xlog and ycurrent > ylog:
            print("NE")
        elif xcurrent < xlog and ycurrent > ylog:
            print("NO")
        elif xcurrent < xlog and ycurrent < ylog:
            print("SO")
        else:
            print("SE")

File: Beecrowd/2-AD-HOC/1091-Divisão_da_Nlogônia-1/1091.js
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

let lines = [];
readline.on('line', (input) => lines.push(input)).on('close', () => {
    let index = 0;

    while (true) {
        let repeat = parseInt(lines[index++]);
        if (repeat === 0) break;

        let [xlog, ylog] = lines[index++].split(' ').map(Number);

        for (let i = 0; i < repeat; i++) {
            let [xcurrent, ycurrent] = lines[index++].split(' ').map(Number);

            if (xcurrent === xlog || ycurrent === ylog) console.log("divisa");
            else if (xcurrent > xlog && ycurrent > ylog) console.log("NE");
            else if (xcurrent < xlog && ycurrent > ylog) console.log("NO");
            else if (xcurrent < xlog && ycurrent < ylog) console.log("SO");
            else console.log("SE");
        }
    }
});
